src/ios/CDVLocation.m

```diff
@@ -191,40 +216,46 @@ - (void)locationManager:(CLLocationManager*)manager
 - (void)getLocation:(CDVInvokedUrlCommand*)command
 {
     [self.commandDelegate runInBackground:^{
+        __weak __typeof(self) weakSelf = self;
         NSString* callbackId = command.callbackId;
         BOOL enableHighAccuracy = [[command argumentAtIndex:0] boolValue];
 
-        if ([self isLocationServicesEnabled] == NO) {
-            NSMutableDictionary* posError = [NSMutableDictionary dictionaryWithCapacity:2];
-            [posError setObject:[NSNumber numberWithInt:PERMISSIONDENIED] forKey:@"code"];
-            [posError setObject:@"Location services are disabled." forKey:@"message"];
-            CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:posError];
-            [self.commandDelegate sendPluginResult:result callbackId:callbackId];
-        } else {
-            if (!self.locationData) {
-                self.locationData = [[CDVLocationData alloc] init];
-            }
-            CDVLocationData* lData = self.locationData;
-            if (!lData.locationCallbacks) {
-                lData.locationCallbacks = [NSMutableArray arrayWithCapacity:1];
-            }
+        [self isLocationServicesEnabledWithCompletion:^(BOOL enabled) {
+            __strong __typeof(weakSelf) strongSelf = weakSelf;
+            
+            if (enabled == NO) {
+                NSMutableDictionary* posError = [NSMutableDictionary dictionaryWithCapacity:2];
+                [posError setObject:[NSNumber numberWithInt:PERMISSIONDENIED] forKey:@"code"];
+                [posError setObject:@"Location services are disabled." forKey:@"message"];
+                CDVPluginResult* result = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsDictionary:posError];
+                [self.commandDelegate sendPluginResult:result callbackId:callbackId];
```

Comment:
Couldn't we use `returnLocationError:withMessage:` here? I believe it does most of what we're doing here.

Reply:
Sure, and which positionError would we send back?         
        // PositionError.PERMISSION_DENIED = 1;
        // PositionError.POSITION_UNAVAILABLE = 2;
        // PositionError.TIMEOUT = 3;

[self returnLocationError: PositionError.? withMessage:@"Location services are disabled."];

Reply:
isn't it there, on line 227?

Reply:
Updated.